core/src/types/error.rs

```diff
@@ -426,6 +428,16 @@ impl From<Error> for io::Error {
     }
 }
 
+impl From<task::JoinError> for Error {
```

Comment:
We should not impl this. Users could convert tokio error to Error wrongly. Instead, we should have a new_tokio_error util function.

Reply:
Actually, I do not get the point. 

You mean the people will the Error in OpenDAL to a place different but related to tokio::tasks:JoinError?

Reply:
Implement From<xxx> for Error could make our developers create opendal error unexpectedly via `?` which may lack of enough context.